torch/_higher_order_ops/effects.py

```diff
@@ -26,6 +26,7 @@ class _EffectType(Enum):
 SIDE_EFFECTS: Dict[OpType, _EffectType] = {
     torch.ops.aten._print.default: _EffectType.ORDERED,
     call_torchbind: _EffectType.ORDERED,
+    torch.ops._c10d_functional.wait_tensor.default: _EffectType.ORDERED,
```

Comment:
We should also delete the two lines here, https://github.com/pytorch/pytorch/blob/main/torch/distributed/_functional_collectives.py#L965